pkg/specgen/generate/kube/kube.go

```diff
@@ -511,45 +515,83 @@ func parseMountPath(mountPath string, readOnly bool, propagationMode *v1.MountPr
 	return dest, opts, nil
 }
 
+func probeToHealthConfig(probe *v1.Probe) (*manifest.Schema2HealthConfig, error) {
+	var commandString string
+	failureCmd := "exit 1"
+	probeHandler := probe.Handler
+
+	// configure healthcheck on the basis of Handler Actions.
+	switch {
+	case probeHandler.Exec != nil:
+		cmd, err := json.Marshal(probeHandler.Exec.Command)
+		if err != nil {
+			return nil, err
+		}
+		commandString = string(cmd)
+	case probeHandler.HTTPGet != nil:
```

Comment:
hm this feels like an imperfect fit. aren't HTTP probes called from the host directly. adding it to commandString would make it an oci runtime exec call inside of the container. While it would achieve the same, it would add the overhead of the exec call that http probes aim to avoid. 

If I were to state my preference, I would say we should wait to support this until we can do it properly. But if it's desireable enough of a feature and the complexity is too high to do correctly, we could go this route, but should document the differences in both the man page and possibly by logging

Reply:
oh nevermind, you're just moving code. oh well